File: chronos-historical-data-service/src/main/java/net/fvogel/chronos/data/model/dto/DataResponseDTO.java
package net.fvogel.chronos.data.model.dto;

import com.fasterxml.jackson.annotation.JsonInclude;
import lombok.Data;
import net.fvogel.chronos.data.model.internal.Entry;
import net.fvogel.chronos.data.model.internal.Relation;

import java.util.ArrayList;
import java.util.List;

@Data
@JsonInclude(JsonInclude.Include.NON_NULL)
public class DataResponseDTO {
    DataResponseMetaInfoDTO meta = new DataResponseMetaInfoDTO();
    List<Entry> entries = new ArrayList<>();
    List<Relation> relations = new ArrayList<>();
}

File: chronos-frontend/src/app/common/model/data/response/data-response.dto.ts
import { EntryDTO } from "./entry.dto";
import { DataResponseMetaInfoDTO } from "./data-response-meta-info.dto";
import { RelationDTO } from "./relation.dto";

export interface DataResponseDTO {
    meta: DataResponseMetaInfoDTO;
    entries: EntryDTO[];
    relations: RelationDTO[];
}